spacy_experimental/span_finder/span_finder_component.py

```diff
@@ -0,0 +1,298 @@
+from typing import List, Dict, Callable, Tuple, Optional, Iterable, Any
+from thinc.api import Config, Model, set_dropout_rate
+from thinc.api import Optimizer
+from thinc.types import Floats2d
+from numpy import float32
+
+from spacy.language import Language
+from spacy.util import registry
+from spacy.pipeline.trainable_pipe import TrainablePipe
+from spacy.tokens import Doc, Token
+from spacy.training import Example
+from spacy.scorer import PRFScore
+
+span_finder_default_config = """
+[model]
+@architectures = "spacy-experimental.span_finder_model.v1"
+
+[model.scorer]
+@layers = "spacy.LinearLogistic.v1"
+nO = 2
+
+[model.tok2vec]
+@architectures = "spacy.Tok2Vec.v1"
+
+[model.tok2vec.embed]
+@architectures = "spacy.MultiHashEmbed.v1"
+width = 96
+rows = [5000, 2000, 1000, 1000]
+attrs = ["ORTH", "PREFIX", "SUFFIX", "SHAPE"]
+include_static_vectors = false
+
+[model.tok2vec.encode]
+@architectures = "spacy.MaxoutWindowEncoder.v1"
+width = ${model.tok2vec.embed.width}
+window_size = 1
+maxout_pieces = 3
+depth = 4
+"""
+
+DEFAULT_SPAN_FINDER_MODEL = Config().from_str(span_finder_default_config)["model"]
+DEFAULT_CANDIDATES_KEY = "span_candidates"
+
+
+@Language.factory(
+    "experimental_span_finder",
+    assigns=["doc.spans"],
+    default_config={
+        "threshold": 0.3,
+        "model": DEFAULT_SPAN_FINDER_MODEL,
+        "candidates_key": DEFAULT_CANDIDATES_KEY,
+        "scorer": {
+            "@scorers": "spacy-experimental.span_finder_scorer.v1",
+            "candidates_key": DEFAULT_CANDIDATES_KEY,
+        },
+    },
+    default_score_weights={
+        "span_finder_f": 1.0,
+        "span_finder_p": 0.0,
+        "span_finder_r": 0.0,
+    },
+)
+def make_span_finder(
+    nlp: Language,
+    name: str,
+    model: Model[List[Doc], Floats2d],
+    scorer: Optional[Callable],
+    threshold: float,
+    candidates_key: str = DEFAULT_CANDIDATES_KEY,
+) -> "SpanFinder":
+    """Create a SpanFinder component. The component predicts whether a token is the start or the end of a potential span.
+    model (Model[List[Doc], Floats2d]): A model instance that
+        is given a list of documents and predicts a probability for each token.
+    threshold (float): Minimum probability to consider a prediction positive.
+    """
+    return SpanFinder(
+        nlp.vocab,
+        model=model,
+        threshold=threshold,
+        name=name,
+        scorer=scorer,
+        candidates_key=candidates_key,
+    )
+
+
+@registry.scorers("spacy-experimental.span_finder_scorer.v1")
```

Comment:
Because we have them as entry points in `setup.cfg`, we can remove the decorators from these functions (also to ensure that the names don't go out-of-sync)